content/docs/what-is-kudo.md

```diff
@@ -133,3 +133,38 @@ Normally, _Operator_ and _OperatorVersion_ are handled by the manager and rarely
+
+```bash
+$ kubectl kudo plan trigger --name deploy --instance my-instance
+```
+
+which will execute `deploy` plan on `my-instance`. While this looks relatively easy on the surface, the devil is in detail, so let's take a closer look.
+
+### Plan Life Cycle
+
+Having the ability to trigger multiple plans on demand raises the question: what happens if two plans run concurrently? The answer is: it depends. Should two plans be completely independent of each other (e.g. `deploy` deploys the services and `monitor` deploys the monitoring pods) both can be executed in parallel. But if two plans in question are `backup` and `restore` or `deploy` and `migrate`? Some plans are incompatible with others. A few may not even be compatible with themselves. While it is probably ok to restart a running `deploy` plan, a `restore` plan might not be reentrant because of possible data corruption.
+
+We're planning to explore the realm of plan compatibility further in the future. Annotating plan affinity and anti-affinity, reentrant vs non-reentrant plans, plan cancelation and transient plan parameters are some of the topics we're examining. All contributions and feedback are highly welcome.
+
+Having all this in mind how can we ensure correct plan execution? Meet Kubernetes admission controllers.
+
+### Admission Controllers
+
+In a nutshell, Kubernetes admission controllers are plugins that govern and enforce how the cluster is used. They can be thought of as a gatekeeper that intercepts (authenticated) API requests and may change the request object or deny the request altogether. Kubernetes already comes with a bunch of these [perinstalled](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/) which govern everything from user authorization to the namespace life cycle.
+
+KUDO manager employs an Instance admission controller that governs changes to Instances, making sure that plans do not interfere. Schematically this looks like the following:
+![Instance update](/images/admission-instance-update.png)
+
+Instance admission controller governs any update to the Instance either through manual plan execution or Instance parameter updates. The general rule of thumb is the following: a plan should be terminal (either successfully with status `COMPLETE` or unsuccessfully with `FATAL_ERROR`) before a new plan is allowed to start. A singular plan can be restarted so we assume all plans to be reentrant at least for now. While this might not be true for all plans, we think that it covers the 80/20 case e.g. when a `deploy` plan is stuck and must be restarted with less memory per node. In case a request is rejected, the Instance controller returns an error explaining why exactly the update was denied.
```

Comment:
```suggestion
Instance admission controller governs any update to the Instance either through manual plan execution or Instance parameter updates. The general rule of thumb is the following: all plans should be terminal (either successfully with status `COMPLETE` or unsuccessfully with `FATAL_ERROR`) before a new plan is allowed to start. A singular plan can be restarted so we assume all plans to be reentrant at least for now. While this might not be true for all plans, we think that it covers the 80/20 case e.g. when a `deploy` plan is stuck and must be restarted with less memory per node. In case a request is rejected, the Instance controller returns an error explaining why exactly the update was denied.
```